site/faq.md

```diff
@@ -157,19 +157,18 @@ There are exceptions:
  - One way of supplying credentials in Kubernetes is to put them on each
    node; Flux does not have access to those credentials.
  - In some environments, authorisation provided by the platform is
-   used instead of image pull secrets. Google Container Registry works
-   this way, for example (and we have introduced a special case for it
-   so Flux will work there too with image pull secrets). See below regarding ECR.
-
-To work around the exceptional cases, you can mount a docker config into
-the Flux container. See the argument `--docker-config` in
-[the daemon arguments reference](https://github.com/weaveworks/flux/blob/master/site/daemon.md#flags).
-
-For ECR (Elastic Container Registry), the credentials supplied by the
-environment are rotated, so it's not possible to supply a file with
-credentials ahead of time. See
-[weaveworks/flux#539](https://github.com/weaveworks/flux/issues/539) for
-workarounds.
+   used instead of image pull secrets:
+    - Google Container Registry works this way; Flux will
+      automatically attempt to use platform-provided credentials when
+      scanning images in GCR.
+    - (Amazon) Elastic Container Registry has its own authentication
```

Comment:
I would change this to
```
Amazon Elastic Container Registry (ECR) has its own authentication; Flux will attempt to scan for images in ECR, if EC2 instance it is running on has access to ECR.
```
Which also suggest that we might want to consider to support for ECR outside of EC2, but I don't think kubelet supports that kind of mode yet (IIRC, it was mentioned in the comment in kubelet code linked from here).

As far as EC2 itself is concerned, there is a possibility that user may wish to avoid using instance roles and pass credentials to flux directly (but we should check if kubelet supports that also). The reason someone may wish to do that is security, i.e. for the lack of real IAM roles for pods, someone may prefer to disallow access to ECR for any but the chose process in their cluster.

Just a few thoughts, perhaps we could open an issue for now.

Reply:
I _think_ it would be possible (though ungainly) to use credentials on the host to grant kubelet access to ECR when not running in EC2. And, I think it'll be possible (with this PR) to make credentials available to flux too, by providing a `~/.aws/config`.

But I'd expect this to be well off the beaten path. I agree that it's important to set out the caveats early -- I'll try to rephrase.